orchagent/lagid.cpp

```diff
@@ -0,0 +1,105 @@
+#include "lagid.h"
+
+LagIdAllocator::LagIdAllocator(
+        _In_ DBConnector* chassis_app_db)
+{
+    SWSS_LOG_ENTER();
+
+    m_dbConnector = chassis_app_db;
+
+    // Load lua script to allocate system lag id. This lua script ensures allocation
+    // of unique system lag id from global chassis app db in atomic fashion when allocation
+    // is requested by different asic instances simultaneously
+
+    string luaScript = loadLuaScript("lagids.lua");
+    m_shaLagId = loadRedisScript(m_dbConnector, luaScript);
+}
+
+int32_t LagIdAllocator::lagIdAdd(
+        _In_ const string &pcname,
+        _In_ int32_t lag_id)
+{
+    SWSS_LOG_ENTER();
+
+    vector<string> keys;
+    keys.push_back(CHASSIS_APP_LAG_ID_START_NAME);
```

Comment:
Avoid taking redis data structure key name while invoking add/del/get operation. The keys can be internally hardcoded

Reply:
Fixed